### PR TITLE
[REL-3063] Update releases terms and recent access changes

### PR DESCRIPTION
**Pull request by cperez08**
# Description
Update some recently changed releases terminology like environment integration and reflect updated permissions on creating environment integrations and integration tokens.

# Reasons
https://circleci.atlassian.net/browse/REL-3063

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [X] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by sds0307 — APPROVED** [review]

File: jekyll/_cci2/roles-and-permissions-overview.adoc [hunk 1]
```diff
@@ -291,32 +291,32 @@ The table below shows the permissions associated with each CircleCI organization
...
 ^| icon:check-circle[]
 
-^| Create environment token
+^| Create integration token
+^| icon:check-circle[]
 ^| icon:check-circle[]
-^|
 ^|
 
-^| Revoke environment token
```
Comment (avalcepina) [inline jekyll/_cci2/roles-and-permissions-overview.adoc:314]: Shouldn't this be "environment integration token"?

File: jekyll/_cci2/roles-and-permissions-overview.adoc [hunk 2]
```diff
@@ -291,32 +291,32 @@ The table below shows the permissions associated with each CircleCI organization
...
+^| Revoke integration token
 ^| icon:check-circle[]
 ^|
 ^|
 
-^| List environment token
```
Comment (avalcepina) [inline jekyll/_cci2/roles-and-permissions-overview.adoc:319]: same here

File: jekyll/_cci2/roles-and-permissions-overview.adoc (hunk 1 above)
Comment (cperez08) [reply jekyll/_cci2/roles-and-permissions-overview.adoc:314]: based on this ticket https://circleci.atlassian.net/browse/REL-3047 should be an integration token.

it was renamed before to that. But I will wait anyways @dlayci  input before merging.

Comment (dlayci) [reply jekyll/_cci2/roles-and-permissions-overview.adoc:314]: For consistency I would agree - we should call it "Environment integration token" everywhere. 

I figured on the environment integration settings page there is enough context there to know it's for the env. integration - but this is lost outside of this context.

**Review by dlayci — APPROVED** [review]

**Review by rosieyohannan — APPROVED** [review]

**Pull request merged**